test/testdata/infer/ruby3_keyword_args.rb

```diff
@@ -11,3 +11,224 @@ def takes_kwargs(x, y:)
 takes_kwargs(99, arghash) # error: Keyword argument hash without `**` is deprecated
 
 takes_kwargs(99, **arghash)
+
+sig {params(h: T::Hash[T.untyped, T.untyped], kw: T::Hash[T.untyped, T.untyped]).void}
+def takes_only_hash_and_kwargs(h, **kw); end
+
+takes_only_hash_and_kwargs(a: "x")
+                        #  ^^^^^^ error: Passing the keyword argument as the last hash parameter is deprecated
```

Comment:
I have two concerns with this:

1. This error message is different from what the Ruby VM prints. On this snippet, the Ruby VM says "wrong number of arguments (given 0, expected 1)"
2. It seems that the way we're reporting this error makes this do something weird:

```ruby
def foo(h, a:, **kw); end
foo(a: "x")
```

Ruby: `wrong number of arguments (given 0, expected 1; required keyword: a)`
this PR: `Missing required keyword argument a for method`

Reply:
>This error message is different from what the Ruby VM prints. On this snippet, the Ruby VM says "wrong number of arguments (given 0, expected 1)"

Hmm, this is weird, I'm not seeing this. Which version of ruby are you using? It should report that error only in ruby 3. This is what I see

```bash
stripe/sorbet [iz/ruby-3-kwargs●] » cat /tmp/x.rb
# typed: true
# experimental-ruby3-keyword-args: true

require 'sorbet-runtime'
extend T::Sig
Warning[:deprecated] = true
sig {params(h: T::Hash[T.untyped, T.untyped], kw: T::Hash[T.untyped, T.untyped]).void}
def takes_only_hash_and_kwargs(h, **kw); end

takes_only_hash_and_kwargs(a: "x")


stripe/sorbet [iz/ruby-3-kwargs●] » ruby --version
ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [arm64-darwin21]


stripe/sorbet [iz/ruby-3-kwargs●] » ruby /tmp/x.rb
/Users/iz/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/sorbet-runtime-0.5.10201/lib/types/private/methods/call_validation.rb:161: warning: Passing the keyword argument as the last hash parameter is deprecated
/tmp/x.rb:8: warning: The called method `takes_only_hash_and_kwargs' is defined here
```

Reply:
```ruby
def foo(h, a:, **kw); end
foo(a: "x")
```
that looks like invalid ruby in 2.7

```bash
stripe/sorbet [iz/ruby-3-kwargs●] » cat /tmp/x.rb
# typed: true
# experimental-ruby3-keyword-args: true

require 'sorbet-runtime'
extend T::Sig
Warning[:deprecated] = true

def foo(h, a:, **kw); end
foo(a: "x")

stripe/sorbet [iz/ruby-3-kwargs●] » ruby /tmp/x.rb
/tmp/x.rb:9: warning: Passing the keyword argument as the last hash parameter is deprecated
/tmp/x.rb:8: warning: The called method `foo' is defined here
Traceback (most recent call last):
        1: from /tmp/x.rb:9:in `<main>'
/tmp/x.rb:8:in `foo': missing keyword: :a (ArgumentError)
```

Reply:
And here is what we report from PR:

```
stripe/sorbet [iz/ruby-3-kwargs●] » sr --experimental-ruby3-keyword-args /tmp/x.rb
Exception::raise(): core/types/subtyping.cc:1076 enforced condition t1 != nullptr has failed: (no message provided)

Backtrace:
...

/tmp/x.rb:9: Failed to type (backtrace is above) https://srb.help/1001
     9 |foo(a: "x")
        ^^^^^^^^^^^

/tmp/x.rb:4: Failed to process tree (backtrace is above) https://srb.help/1001
     4 |require 'sorbet-runtime'
     5 |extend T::Sig
     6 |Warning[:deprecated] = true
     7 |
     8 |def foo(h, a:, **kw); end
     9 |foo(a: "x")

/tmp/x.rb: Exception in cfg+infer: /tmp/x.rb (backtrace is above) https://srb.help/1001
Errors: 3
```

I'll look into that crash